### ISSUE TITLE
[2.44] ArrayIndexOfOutBoundsException

### ISSUE THREAD

**Issue by tchegito** [tchegito/zildo#137]
Exception during sprite ordering. How could we have so much sprites ???

```
map=sousbois1 (42x45)
sprites=[Perso=Zildo
Coords:(594.1206, 102.04857 0.0)
info=ZILDO
mvt=VIDE pv=9 weapon=[FLUT,0] name=Roxy lasthit=null, 596.1206, 112.04857 (80 - bank 0), 691.0, 540.0 (83 - bank 0), 596.1206, 102.04857 (SHADOW_SMALL), 316.8378, 261.85196 (103 - bank 0), Perso=blob1
Coords:(419.0, 362.0 0.0) ==> (419,362)
info=ENEMY
mvt=VIDE pv=2, 0.0, 0.0 (80 - bank 0), 0.0, 0.0 (SHADOW), Perso=isidore
Coords:(456.0, 581.0 0.0)
info=NEUTRAL
mvt=VIDE pv=1, 0.0, 0.0 (80 - bank 0), 456.0, 580.0 (SHADOW)]
persos=[Perso=Zildo
Coords:(594.1206, 102.04857 0.0)
info=ZILDO
mvt=VIDE pv=9 weapon=[FLUT,0] name=Roxy lasthit=null, Perso=blob1
Coords:(419.0, 362.0 0.0) ==> (419,362)
info=ENEMY
mvt=VIDE pv=2, Perso=isidore
Coords:(456.0, 581.0 0.0)
info=NEUTRAL
mvt=VIDE pv=1]
variables={Dizzie=[[DYNAMITE, 1], 15, 20], allowedDynamite=yes, BilelIgorVillage=[[DYNAMITE,1],100,0], fishWater=6.0, nettleCount=-1, Carlo=[[MOON, 1], 800, 1], moneyStolen=322.0}
scripts=0 scripts running {[}
lastdialog=Lugdunia
Elle a passé la nuit à errer dans les sous-bois. Mais voyons ce qui lui est arrivé.
Il s'agit bien sûr de la princesse Roxy ! Que lui est-il arrivé depuis sa transformation en écureuil ?
Cependant, il est un personnage que nous avons peu vu lors de l'épisode 2 et qui est pourtant central.
Plus que jamais, le destin du Royaume était entre les mains de Roxy.
Les gardes noirs ne seraient alors plus invincibles, tant que le nouveau Héros de Lugdunia serait prêt au combat.
Igor ralluma la forge, et était ainsi prêt à perfectionner l'épée de Roxy.
Château de Lugdunia
Tout naturellement, ils le logèrent dans une maison qui n'attendait que lui.
Roxy et Hector avaient délivré Igor, et ils le ramenèrent à Lugdunia, sain et sauf.
�Igor�: Avec plaisir ! Filons d'ici.
�Jerem�: Oui, suivez moi vite. Il y a un passage par ici.
�Igor�: Ah ! Te revoilà !
Grotte de Lugdunia
�Hector�: Va chercher Igor, et reviens vite, Roxy.
�Hector�: En haut de cette échelle, tu te trouveras devant la prison, là où je t'ai parlé tout à l'heure.
�Hector�: Voilà, nous y sommes.
�Hector�: Apparemment, les gardes noirs ignorent l'existence de ce passage, car aucun d'eux n'a cru bon de le surveiller.
�Hector�: Elle nous a permis d'amener le peu de troupes qu'il nous restait pendant la nuit et au petit matin, avec l'effet de surprise, nous avons libéré la ville.
�Hector�: Tu sais, Roxy, cette grotte a été creusée pendant la guerre des 2 royaumes, lorsque Farabrume a pris possession de notre château.
�Hector�: Suis moi, ce passage nous mène tout droit dans l'enceinte du château.
Lugdunia
�Hector�: Suis moi Roxy.
�Hector�: Formidable ! Nous le ramènerons par le passage secret.
java.lang.ArrayIndexOutOfBoundsException: length=500; index=500
	at zildo.monde.sprites.utils.SpriteSorter.orderSpritesByBank(SpriteSorter.java:211)
	at zildo.client.SpriteDisplay.updateSpritesClient(SpriteDisplay.java:169)
	at zildo.client.ClientEngineZildo.renderFrame(ClientEngineZildo.java:179)
	at zildo.platform.opengl.AndroidOpenGLGestion.render(AndroidOpenGLGestion.java:406)
	at zildo.client.Client.render(Client.java:273)
	at zildo.client.Client.mainLoop(Client.java:233)
	at com.alembrum.OpenGLRenderer.onDrawFrame(OpenGLRenderer.java:107)
	at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1582)
	at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1267)
```

**Comment by tchegito** [comment]
Got it: at the beginning of the third episode, there's a title (called 'synthe' in code) "Episode 3 : Rising power". When it is displayed, we have two bugs:
- compass menu leads to inifinte blink
- ingame menu doesn't disappear: it's overlapping title

**Idea:** forbid ingame and compass menu during Synthe. In any case, this would cause too much overlapping fonts, and text would be unreadable.

**Comment by tchegito** [comment]
[Fixed in 2.45]

3 improvements:
- ESC/BACK/MENU is forbidden during TitleStage
- DialogMode gets back to CLASSIC when TitleStage is over
- compass stay visible when hero try to talk to a character who has nothing to say (for instance Igor in front of the prison)

c78f3e3db40d3c15747369d808dfb81b7f842681
